apps/namadillo/src/utils/index.ts

```diff
@@ -38,32 +39,46 @@ export const proposalIdToString = (proposalId: bigint): string =>
 
 export const useTransactionEventListener = <T extends keyof WindowEventMap>(
   event: T,
-  handler: (this: Window, ev: WindowEventMap[T]) => void,
-  deps: React.DependencyList = []
+  handler: (event: WindowEventMap[T]) => void
 ): void => {
+  // `handlerRef` is useful to avoid recreating the listener every time
+  const handlerRef = useRef(handler);
+  handlerRef.current = handler;
+
   useEffect(() => {
-    window.addEventListener(event, handler);
+    const callback: typeof handler = (event) => handlerRef.current(event);
+    window.addEventListener(event, callback);
     return () => {
-      window.removeEventListener(event, handler);
+      window.removeEventListener(event, callback);
     };
-  }, deps);
+  }, [event]);
```

Comment:
I lost many hours debugging until realizing this dependencies was the issue 😢 

Maybe we should enable the `"react-hooks/exhaustive-deps": "warn"` and, if you don't want to use, at least you need to add the lint ignore comment

What do you think?

Reply:
we don't need the `deps` anymore, because the function is saved on the `ref`, so you are free to pass anything, even a new function everytime